crates/prune/src/segments/mod.rs

```diff
@@ -57,16 +63,42 @@ pub trait Segment<DB: Database>: Debug + Send + Sync {
     ) -> ProviderResult<()> {
         provider.save_prune_checkpoint(self.segment(), checkpoint)
     }
+
+    /// Gets the limiter for this specific segment, built from the limiter used in the parent
+    /// scope.
+    fn new_limiter_from_parent_scope_limiter(&self, limiter: &PruneLimiter) -> PruneLimiter {
+        PruneLimiterBuilder::floor_deleted_entries_limit_to_multiple_of(
+            limiter,
+            NonZeroUsize::new(1).unwrap(),
+        )
+        .build()
+    }
 }
 
 /// Segment pruning input, see [Segment::prune].
-#[derive(Debug, Clone, Copy)]
+#[derive(Debug)]
 pub struct PruneInput {
     pub(crate) previous_checkpoint: Option<PruneCheckpoint>,
     /// Target block up to which the pruning needs to be done, inclusive.
     pub(crate) to_block: BlockNumber,
-    /// Maximum entries to delete from the database.
-    pub(crate) delete_limit: usize,
+    /// Limits pruning of segment in prune job.
+    pub(crate) limiter: PruneLimiter,
+}
+
+impl Clone for PruneInput {
+    fn clone(&self) -> Self {
+        let Self { previous_checkpoint, to_block, limiter } = self;
+
+        let previous_checkpoint = *previous_checkpoint;
+        let to_block = *to_block;
+        let limiter = PruneLimiterBuilder::floor_deleted_entries_limit_to_multiple_of(
+            limiter,
+            NonZeroUsize::new(1).unwrap(),
+        )
+        .build();
+
+        Self { previous_checkpoint, to_block, limiter }
+    }
```

Comment:
IMO we shouldn't hide this behaviour inside the clone, and it should be done explicitly